source/client/output_formatter_impl.cc

```diff
@@ -138,5 +144,119 @@ DottedStringOutputFormatterImpl::formatProto(const nighthawk::client::Output& ou
   return ss.str();
 }
 
+const nighthawk::client::Result&
+FortioOutputFormatterImpl::getGlobalResult(const nighthawk::client::Output& output) const {
+  for (const auto& nh_result : output.results()) {
+    if (nh_result.name() == "global") {
+      return nh_result;
+    }
+  }
+
+  throw NighthawkException("Nighthawk output was malformed, contains no 'global' results.");
+}
+
+const nighthawk::client::Counter&
+FortioOutputFormatterImpl::getCounterByName(const nighthawk::client::Result& result,
+                                            absl::string_view counter_name) const {
+  for (const auto& nh_counter : result.counters()) {
+    if (nh_counter.name() == counter_name) {
+      return nh_counter;
+    }
+  }
+
+  throw NighthawkException(absl::StrCat(
+      "Nighthawk result was malformed, contains no counter with name: ", counter_name));
+}
+
+const nighthawk::client::Statistic& FortioOutputFormatterImpl::getRequestResponseStatistic(
+    const nighthawk::client::Result& result) const {
+  for (auto const& nh_stat : result.statistics()) {
+    if (nh_stat.id() == "benchmark_http_client.request_to_response") {
+      return nh_stat;
+    }
+  }
+
+  throw NighthawkException("Nighthawk result was malformed, contains no "
+                           "'benchmark_http_client.request_to_response' statistic.");
+}
+
+std::string FortioOutputFormatterImpl::formatProto(const nighthawk::client::Output& output) const {
+  nighthawk::client::FortioResult fortio_output;
+
+  // TODO(nareddyt): Not needed but nice to have, displays in the UI
+  fortio_output.mutable_labels()->set_value("A random label");
+
+  fortio_output.mutable_starttime()->set_seconds(output.timestamp().seconds());
+  fortio_output.mutable_requestedqps()->set_value(output.options().requests_per_second().value());
+  fortio_output.mutable_requestedduration()->set_seconds(output.options().duration().seconds());
+
+  // FIXME: Set the actual to the same as the requested since NH does not expose these fields
+  fortio_output.mutable_actualqps()->set_value(output.options().requests_per_second().value());
+  fortio_output.mutable_actualduration()->set_value(output.options().duration().seconds());
+
+  // This displays as "connections" in the UI, not threads
+  fortio_output.mutable_numthreads()->set_value(output.options().connections().value());
+
+  // This displays the endpoint URL in the UI
+  fortio_output.mutable_url()->set_value(output.options().uri().value());
+
+  // Get the result that represents all workers (global)
+  const auto& nh_global_result = this->getGlobalResult(output);
+
+  // Fill in the number of successful responses.
+  // Fortio-ui only reads the 200 OK field, other fields are only informational.
+  const auto& nh_2xx_counter = this->getCounterByName(nh_global_result, "benchmark.http_2xx");
```

Comment:
I think it is possible that this counter can not be found in NH's output. For example, when all responses returned 404. (0 statistics are not emitted). I'm not sure how to translate that into the domain of Fortio's json, and perhaps the exception raised in that case is the right thing to do (though mentioning the input being untranslatable might be helpful).

Reply:
I think we can handle this by just setting it to 0.